site/zenodo_rdm/curation/rules.py

```diff
@@ -157,3 +162,87 @@ def award_acronym_in_additional_description(record):
                 ):
                     return True
     return False
+
+
+def eu_community_declined_request(record):
+    """Check if record was rejected from EU community."""
+    community_requests = dsl.Q(
+        "bool",
+        must=[
+            dsl.Q(
+                "term",
+                **{"receiver.community": current_app.config.get("EU_COMMUNITY_UUID")},
+            ),
+            dsl.Q("term", **{"topic.record": record.pid.pid_value}),
+        ],
+    )
+    request_types = dsl.Q(
+        "bool",
+        should=[
+            dsl.Q("term", **{"type": CommunityInclusion.type_id}),
+            dsl.Q("term", **{"type": CommunitySubmission.type_id}),
+        ],
+        minimum_should_match=1,
+    )
+    finalq = community_requests & request_types
+    results = current_requests_service.search(system_identity, extra_filter=finalq)
+
+    for result in results:
+        if result["is_closed"] and result["status"] == "declined":
+            return True
+        if result["is_open"] and not result["is_expired"]:
+            return True
+    return False
+
+
+def eu_subcommunity_declined_request(record):
+    """Check if record was rejected from EU sub community."""
+    record_requests = dsl.Q(
+        "bool",
+        must=[
+            dsl.Q("term", **{"topic.record": record.pid.pid_value}),
+            dsl.Q("term", **{"is_open": False}),
+        ],
+    )
+    request_types = dsl.Q(
+        "bool",
+        should=[
+            dsl.Q("term", **{"type": CommunityInclusion.type_id}),
+            dsl.Q("term", **{"type": CommunitySubmission.type_id}),
+        ],
+        minimum_should_match=1,
+    )
+    finalq = record_requests & request_types
+    results = current_requests_service.search(system_identity, extra_filter=finalq)
+
+    for result in results:
+        receiver = current_communities.service.record_cls.pid.resolve(
+            result["receiver"]["community"]
+        )
+        if receiver.parent and str(receiver.parent.id) == current_app.config.get(
+            "EU_COMMUNITY_UUID"
+        ):
+            if result["status"] == "declined":
+                return True
+    return False
+
+
+def community_name_award_acronym(record):
+    """Check if award acronym in community name."""
+    comm_text = ""
+    for comm in record.parent.communities:
+        comm_text += comm.metadata.get("title", "")
+        comm_text += comm.metadata.get("page", "")
+
+    if comm_text:
+        award_service = current_service_registry.get("awards")
+        funding = record.metadata.get("funding", [])
+        for f in funding:
+            if f["funder"].get("id") == "00k4n6c32":
+                if award_id := f.get("award", {}).get("id"):
+                    award = award_service.record_cls.pid.resolve(award_id)
+                    if award.get("acronym") and (
+                        award.get("acronym").lower() in comm_text.lower()
```

Comment:
minor: I would do some form of word-splitting here to be safer with the acronym match. Otherwise a project with acronym `SHINE` would also be matched in the text "...this approach out**shine**s all other methods...". 

Maybe the same thing should be applied in other rules that check for the acronym?